app/grandchallenge/evaluation/models.py

```diff
@@ -1715,6 +1729,46 @@ def get_absolute_url(self):
             },
         )
 
+    @cached_property
+    def has_matching_algorithm_interfaces(self):
+        return sorted(
+            [int.pk for int in self.phase.algorithm_interfaces.all()]
+        ) == sorted(
+            [int.pk for int in self.algorithm_image.algorithm.interfaces.all()]
+        )
+
+    @cached_property
+    def has_blocking_algorithm_jobs(self):
+        if not self.predictions_file:
+            # prediction submissions never have blocking jobs
+            return False
+        else:
+            # are there non-successful jobs for the algorithm image, model and archive?
+            algorithm_interfaces = self.phase.algorithm_interfaces.all()
+
+            items = get_archive_items_for_interfaces(
+                algorithm_interfaces=algorithm_interfaces,
+                archive_items=self.phase.archive.items.all(),
+            )
+            non_success_statuses = [
+                status[0]
+                for status in Job.STATUS_CHOICES
+                if status[0] != Job.SUCCESS
+            ]
```

Comment:
The issue only mentions failed and cancelled jobs, but I think we want to check for any non-successful jobs as re-evaluating should not be possible if there are queued jobs etc either.